### PR TITLE
feature/client-restaurant/voyager

### PR DESCRIPTION
**Pull request by mosayed01**
test voyager navigation with Koin using Koin Compose version. 
Add base clases

### CODE REVIEW

File: client_restaurant/shared/build.gradle.kts [hunk 1]
```diff
@@ -33,6 +38,13 @@ kotlin {
                 implementation(libs.compose.foundation)
                 implementation(libs.compose.material3)
                 implementation(libs.compose.components.resources)
+
+                implementation(libs.bundles.voyager)
+                implementation(libs.kotlin.coroutines)
+                api(libs.koin.core)
+                implementation(libs.koin.annotations)
+//                implementation(libs.koin.ksp)
```
Comment (ahmednasserzaza) [inline client_restaurant/shared/build.gradle.kts:46]: could you remove this comment

File: client_restaurant/shared/src/commonMain/kotlin/di/AppModule.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import presentation.login.LoginScreenModel
+import presentation.main.MainScreenModel
+
+fun appModule() = module {
+    includes(
+        screenModule,
+    )
+}
+
+val screenModule = module {
```
Comment (ahmednasserzaza) [inline client_restaurant/shared/src/commonMain/kotlin/di/AppModule.kt:14]: should be better to be in a separated file

**Review by ahmednasserzaza — APPROVED** [review]
Well Done 👏

**Review by mustafa-sab3alarab — APPROVED** [review]

**Comment by ahmednasserzaza** [comment]
Don't forget to resolve conflict

**Pull request merged**